tests/test_rank.py

```diff
@@ -403,42 +393,59 @@ def test_get_knn_distance_ood_scores():
     X_test_with_ood = np.vstack([X_test, X_ood])
 
     # Fit nearest neighbors on X_train
-    nbrs = NearestNeighbors(n_neighbors=5).fit(X_train)
+    knn = NearestNeighbors(n_neighbors=5).fit(X_train)
 
     # Get KNN distance as OOD score
     k = 5
-    knn_distance_ood_score = rank.get_knn_distance_ood_scores(
-        features=X_test_with_ood, nbrs=nbrs, k=k
-    )
+    knn_distance_ood_score = rank.get_outlier_scores(features=X_test_with_ood, knn=knn, k=k)
```

Comment:
Add assert after this to verify the k is properly set?  

One failure mode I can see for this code is the default k=10 accidentally somehow overwrites a user-specified value of k.  Current unit tests seems like they'd still pass even if the code internally was using the wrong k value, so would be good to have test where this cannot happen (see also below comment).